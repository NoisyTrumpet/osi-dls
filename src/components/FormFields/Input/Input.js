import React from 'react';
import PropTypes from 'prop-types';
import './Input.scss';
import classNames from 'classnames';

const Input = ({
  ariaControls,
  autoComplete,
  click,
  domId,
  error,
  isChecked,
  isDisabled,
  maxLength,
  name,
  onBlur,
  onChange,
  onFocus,
  placeholder,
  required,
  type,
  value,
  // range input
  min,
  max,
  step,
}) => {
  const classes = classNames(
    'input-field',
    { error },
  );

  return (
    <input
      aria-controls={ariaControls}
      autoComplete={autoComplete || undefined}
      type={type}
      name={name}
      id={domId}
      className={classes}
      maxLength={maxLength}
      placeholder={placeholder}
      defaultValue={value}
      checked={isChecked}
      disabled={isDisabled}
      aria-required={required}
      aria-invalid={error}
      onChange={onChange}
      onBlur={onBlur}
      onClick={click}
      onFocus={onFocus}
      min={min}
      max={max}
      step={step}
    />
  );
};

Input.defaultProps = {
  domId: '',
  name: '',
  onChange: null,
  onBlur: null,
  placeholder: '',
  type: 'text',
  value: '',
};

Input.propTypes = {
  ariaControls: PropTypes.string,
  autoComplete: PropTypes.oneOf(['on', 'off']),
  click: PropTypes.func,
  domId: PropTypes.string,
  error: PropTypes.bool,
  isChecked: PropTypes.bool,
  isDisabled: PropTypes.bool,
  maxLength: PropTypes.number,
  name: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  type: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]),
  min: PropTypes.string,
  max: PropTypes.string,
  step: PropTypes.string,
};

export default Input;
