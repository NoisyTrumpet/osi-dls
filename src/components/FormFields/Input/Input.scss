.input-field {
  color: map-get($colors, 'brand-grey');
  border: rem(2px) solid map-get($colors, 'gray3');
  border-radius: $space-8;
  order: 1;
  padding: $space-12;
  @include fontSans($beam-84, $beam-vf, 'sans-serif', 84, 700);

  &:focus {
    border-color: map-get($colors, 'brand-blue');
    outline: none;
  }

  &[type='checkbox'],
  &[type='radio'] {
    cursor: pointer;
    opacity: 0;
    position: absolute;
    z-index: 1;
    @include appearance(none);
  }

  &[type='checkbox'],
  &[type='radio'] {
    height: rem(18px);
    width: rem(18px);
  }

  &[type='text'] {
    width: 100%;
  }

  &:disabled {
    color: map-get($colors, 'disabled');
    cursor: default;
    border-color: map-get($colors, 'disabled');

    ~ .input-label {
      cursor: default;

      .typography {
        color: map-get($colors, 'disabled');
      }
    }

    ~ .indicator {
      border-color: map-get($colors, 'disabled');

      &:after {
        background-color: map-get($colors, 'disabled');
      }
    }
  }

  &.error {
    border-color: map-get($colors, 'util-error');

    &:focus {
      border-color: map-get($colors, 'util-error');
    }
  }
}

.input-field-text {
  .helper-text {
    padding-left: 15px;
  }
}

.input-field-select {
  .helper-text {
    padding-left: 15px;
  }
}

// container ... normally goes in a forms.scss
.input-field-checkbox,
.input-field-radio {
  align-items: center;
  display: flex;
  position: relative;

  .input-field,
  .input-label {
    &:hover {
      ~ .indicator {
        background-color: map-get($colors, 'gray1');
      }
    }
  }

  .input-field {
    &:checked {
      ~ .indicator {
        background-color: map-get($colors, 'brand-blue');

        &:after {
          opacity: 1;
        }
      }
    }

    &:disabled {
      ~ .indicator {
        background-color: map-get($colors, 'brand-transparent');
      }
    }
  }

  .input-label {
    cursor: pointer;
    line-height: $space-24;
    @include user-select;

    &.isDisabled {
      cursor: auto;
    }
  }
}

.input-field-checkbox {
  .helper-text {
    margin: 0 auto 6px -26px;
  }
}

// Prevent visual toggle when helpertext appears
.no-helpertext {
  margin-bottom: 30px;
}
