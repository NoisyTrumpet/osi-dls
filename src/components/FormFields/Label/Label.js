import React from 'react';
import PropTypes from 'prop-types';
import './Label.scss';
import classNames from 'classnames';
import { Typography } from '../../Typography';

const Label = ({
  fieldType,
  forId,
  display,
  hideLabel,
  isDisabled,
  color,
  required,
  addClass,
  suffixHelperText,
  children,
}) => {
  const classes = classNames(
    'input-label',
    fieldType,
    display,
    hideLabel && 'hide-label',
    addClass,
    { isDisabled },
  );

  const srtVariant = hideLabel && "srt";

  return (
    <label
      className={classes}
      htmlFor={forId}
    >
      <Typography variant={srtVariant || "field-label"} color={color} addClass={addClass}>
        {children}
      </Typography>
      {suffixHelperText && (
        <Typography variant="helper" addClass="optional">
          {suffixHelperText}
        </Typography>
      )}
      {!required && (
        <Typography variant={srtVariant || "helper"} addClass="optional">
          (Optional)
        </Typography>
      )}
    </label>
  );
};

Label.defaultProps = {
  fieldType: 'text',
  display: 'inline',
  isDisabled: false,
  color: '',
};

Label.propTypes = {
  fieldType: PropTypes.oneOf([
    'text',
    'radio',
    'checkbox',
    'textarea',
    'select',
  ]),
  forId: PropTypes.string,
  display: PropTypes.oneOf([
    'inline',
    'inline-block',
    'block',
  ]),
  hideLabel: PropTypes.bool,
  isDisabled: PropTypes.bool,
  color: PropTypes.string,
  suffixHelperText: PropTypes.string, // helper text that is appended to label
  required: PropTypes.bool,
  addClass: PropTypes.string,
  children: PropTypes.node,
};

export default Label;
