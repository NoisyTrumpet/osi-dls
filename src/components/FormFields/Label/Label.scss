.input-label {
  color: map-get($colors, 'gray5');
  padding-left: $space-8;
  order: 2;

  &.block {
    display: block;
  }

  &.inline-block {
    display: inline-block;
  }

  &.text {
    position: absolute;
    top: rem(-11px);
    left: rem(2px);
    transform: translate(0, 50%);

    .optional {
      font-size: rem(12px);
      padding-left: rem(2px);
      padding-right: 0px;
      position: absolute;
      background: #ffffff;
    }
    
    .typography-field-label {
      background: map-get($colors, 'brand-white');
      padding: 0 $space-4;
    }
  }
}
