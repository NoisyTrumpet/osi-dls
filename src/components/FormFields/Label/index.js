export { default as Label } from './Label'; // eslint-disable-line import/prefer-default-export
