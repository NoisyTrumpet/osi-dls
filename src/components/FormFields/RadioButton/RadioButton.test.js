import React from 'react';
import { mount } from 'enzyme';
import { Formik, Form } from 'formik';
import toJson from 'enzyme-to-json';
import { Label } from '../Label';
import { Input } from '../Input';
import { RadioButton } from './';

describe('<RadioButton />', () => {
  it('Should render', () => {
    const wrapper = mount(
      <Formik initialValues={{ radioGroup: '' }}>
        <Form>
          <RadioButton
            label="Radio Btn 1"
            name="radioGroup"
            value="radioBtn1"
            domId="radioBtn1"
          />
          <RadioButton
            label="Radio Btn 2"
            name="radioGroup"
            value="radioBtn2"
            domId="radioBtn2"
          />
        </Form>
      </Formik>,
    );
    expect(wrapper.length).toBe(1);
    expect(wrapper.find(Input).length).toBe(2);
    expect(wrapper.find(Label).length).toBe(2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should render card style', () => {
    const wrapper = mount(
      <Formik initialValues={{ radioGroup: '' }}>
        <Form>
          <RadioButton
            label="Radio Button Card 1"
            name="myRadioGroup"
            domId="myDomId"
            value="Option 1"
            isCard
            helperText="Optional helper text"
            checked
          />
          <RadioButton
            label="Radio Button Card 1"
            name="myRadioGroup"
            domId="myDomId"
            value="Option 1"
            isCard
            helperText="Optional helper text"
            checked
          />
          <RadioButton
            label="Radio Button Card 1"
            name="myRadioGroup"
            domId="myDomId"
            value="Option 1"
            isCard
            helperText="Optional helper text"
            checked
          />
          <RadioButton
            label="Radio Button Card 1"
            name="myRadioGroup"
            domId="myDomId"
            value="Option 1"
            isCard
            helperText="Optional helper text"
            checked
            isDisabled
          />
        </Form>
      </Formik>,
    );
    expect(wrapper.length).toBe(1);
    expect(wrapper.find(Input).length).toBe(4);
    expect(wrapper.find(Label).length).toBe(4);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
