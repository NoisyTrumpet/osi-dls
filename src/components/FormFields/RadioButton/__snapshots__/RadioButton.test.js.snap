// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> Should render 1`] = `
<Formik
  initialValues={
    Object {
      "radioGroup": "",
    }
  }
>
  <Form>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <RadioButton
        domId="radioBtn1"
        isCard={false}
        isDisabled={false}
        label="Radio Btn 1"
        name="radioGroup"
        value="radioBtn1"
      >
        <div
          className="input-field-container input-field-radio"
        >
          <Input
            checked={false}
            className="radio-button"
            domId="radioBtn1"
            isChecked={false}
            isDisabled={false}
            name="radioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="radio"
            value="radioBtn1"
          >
            <input
              checked={false}
              className="input-field"
              defaultValue="radioBtn1"
              disabled={false}
              id="radioBtn1"
              name="radioGroup"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="radio"
            />
          </Input>
          <Indicator
            type="radio"
          >
            <span
              aria-hidden="true"
              className="indicator indicator-radio"
            />
          </Indicator>
          <div
            className="radio-button-labels undefined"
          >
            <Label
              color=""
              display="inline-block"
              fieldType="radio"
              forId="radioBtn1"
              isDisabled={false}
              required={true}
            >
              <label
                className="input-label radio inline-block"
                htmlFor="radioBtn1"
              >
                <Typography
                  color=""
                  variant="field-label"
                >
                  <span
                    className="typography typography-field-label"
                  >
                    Radio Btn 1
                  </span>
                </Typography>
              </label>
            </Label>
          </div>
        </div>
      </RadioButton>
      <RadioButton
        domId="radioBtn2"
        isCard={false}
        isDisabled={false}
        label="Radio Btn 2"
        name="radioGroup"
        value="radioBtn2"
      >
        <div
          className="input-field-container input-field-radio"
        >
          <Input
            checked={false}
            className="radio-button"
            domId="radioBtn2"
            isChecked={false}
            isDisabled={false}
            name="radioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="radio"
            value="radioBtn2"
          >
            <input
              checked={false}
              className="input-field"
              defaultValue="radioBtn2"
              disabled={false}
              id="radioBtn2"
              name="radioGroup"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="radio"
            />
          </Input>
          <Indicator
            type="radio"
          >
            <span
              aria-hidden="true"
              className="indicator indicator-radio"
            />
          </Indicator>
          <div
            className="radio-button-labels undefined"
          >
            <Label
              color=""
              display="inline-block"
              fieldType="radio"
              forId="radioBtn2"
              isDisabled={false}
              required={true}
            >
              <label
                className="input-label radio inline-block"
                htmlFor="radioBtn2"
              >
                <Typography
                  color=""
                  variant="field-label"
                >
                  <span
                    className="typography typography-field-label"
                  >
                    Radio Btn 2
                  </span>
                </Typography>
              </label>
            </Label>
          </div>
        </div>
      </RadioButton>
    </form>
  </Form>
</Formik>
`;

exports[`<RadioButton /> Should render card style 1`] = `
<Formik
  initialValues={
    Object {
      "radioGroup": "",
    }
  }
>
  <Form>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <RadioButton
        checked={true}
        domId="myDomId"
        helperText="Optional helper text"
        isCard={true}
        isDisabled={false}
        label="Radio Button Card 1"
        name="myRadioGroup"
        value="Option 1"
      >
        <div
          className="input-field-container input-field-radio input-field-radio-card"
        >
          <Input
            checked={true}
            className="radio-button"
            domId="myDomId"
            isChecked={false}
            isDisabled={false}
            name="myRadioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="radio"
            value="Option 1"
          >
            <input
              checked={false}
              className="input-field"
              defaultValue="Option 1"
              disabled={false}
              id="myDomId"
              name="myRadioGroup"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="radio"
            />
          </Input>
          <Indicator
            type="radio"
          >
            <span
              aria-hidden="true"
              className="indicator indicator-radio"
            />
          </Indicator>
          <div
            className="radio-button-labels undefined"
          >
            <Label
              color=""
              display="inline-block"
              fieldType="radio"
              forId="myDomId"
              isDisabled={false}
              required={true}
            >
              <label
                className="input-label radio inline-block"
                htmlFor="myDomId"
              >
                <Typography
                  color=""
                  variant="field-label"
                >
                  <span
                    className="typography typography-field-label"
                  >
                    Radio Button Card 1
                  </span>
                </Typography>
              </label>
            </Label>
            <HelperText>
              <div
                className="helper-text"
              >
                <Typography
                  variant="helper"
                >
                  <span
                    className="typography typography-helper"
                  >
                    Optional helper text
                  </span>
                </Typography>
              </div>
            </HelperText>
          </div>
        </div>
      </RadioButton>
      <RadioButton
        checked={true}
        domId="myDomId"
        helperText="Optional helper text"
        isCard={true}
        isDisabled={false}
        label="Radio Button Card 1"
        name="myRadioGroup"
        value="Option 1"
      >
        <div
          className="input-field-container input-field-radio input-field-radio-card"
        >
          <Input
            checked={true}
            className="radio-button"
            domId="myDomId"
            isChecked={false}
            isDisabled={false}
            name="myRadioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="radio"
            value="Option 1"
          >
            <input
              checked={false}
              className="input-field"
              defaultValue="Option 1"
              disabled={false}
              id="myDomId"
              name="myRadioGroup"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="radio"
            />
          </Input>
          <Indicator
            type="radio"
          >
            <span
              aria-hidden="true"
              className="indicator indicator-radio"
            />
          </Indicator>
          <div
            className="radio-button-labels undefined"
          >
            <Label
              color=""
              display="inline-block"
              fieldType="radio"
              forId="myDomId"
              isDisabled={false}
              required={true}
            >
              <label
                className="input-label radio inline-block"
                htmlFor="myDomId"
              >
                <Typography
                  color=""
                  variant="field-label"
                >
                  <span
                    className="typography typography-field-label"
                  >
                    Radio Button Card 1
                  </span>
                </Typography>
              </label>
            </Label>
            <HelperText>
              <div
                className="helper-text"
              >
                <Typography
                  variant="helper"
                >
                  <span
                    className="typography typography-helper"
                  >
                    Optional helper text
                  </span>
                </Typography>
              </div>
            </HelperText>
          </div>
        </div>
      </RadioButton>
      <RadioButton
        checked={true}
        domId="myDomId"
        helperText="Optional helper text"
        isCard={true}
        isDisabled={false}
        label="Radio Button Card 1"
        name="myRadioGroup"
        value="Option 1"
      >
        <div
          className="input-field-container input-field-radio input-field-radio-card"
        >
          <Input
            checked={true}
            className="radio-button"
            domId="myDomId"
            isChecked={false}
            isDisabled={false}
            name="myRadioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="radio"
            value="Option 1"
          >
            <input
              checked={false}
              className="input-field"
              defaultValue="Option 1"
              disabled={false}
              id="myDomId"
              name="myRadioGroup"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="radio"
            />
          </Input>
          <Indicator
            type="radio"
          >
            <span
              aria-hidden="true"
              className="indicator indicator-radio"
            />
          </Indicator>
          <div
            className="radio-button-labels undefined"
          >
            <Label
              color=""
              display="inline-block"
              fieldType="radio"
              forId="myDomId"
              isDisabled={false}
              required={true}
            >
              <label
                className="input-label radio inline-block"
                htmlFor="myDomId"
              >
                <Typography
                  color=""
                  variant="field-label"
                >
                  <span
                    className="typography typography-field-label"
                  >
                    Radio Button Card 1
                  </span>
                </Typography>
              </label>
            </Label>
            <HelperText>
              <div
                className="helper-text"
              >
                <Typography
                  variant="helper"
                >
                  <span
                    className="typography typography-helper"
                  >
                    Optional helper text
                  </span>
                </Typography>
              </div>
            </HelperText>
          </div>
        </div>
      </RadioButton>
      <RadioButton
        checked={true}
        domId="myDomId"
        helperText="Optional helper text"
        isCard={true}
        isDisabled={true}
        label="Radio Button Card 1"
        name="myRadioGroup"
        value="Option 1"
      >
        <div
          className="input-field-container input-field-radio input-field-radio-card input-field-disabled"
        >
          <Input
            checked={true}
            className="radio-button"
            domId="myDomId"
            isChecked={false}
            isDisabled={true}
            name="myRadioGroup"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="radio"
            value="Option 1"
          >
            <input
              checked={false}
              className="input-field"
              defaultValue="Option 1"
              disabled={true}
              id="myDomId"
              name="myRadioGroup"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="radio"
            />
          </Input>
          <Indicator
            type="radio"
          >
            <span
              aria-hidden="true"
              className="indicator indicator-radio"
            />
          </Indicator>
          <div
            className="radio-button-labels undefined"
          >
            <Label
              color=""
              display="inline-block"
              fieldType="radio"
              forId="myDomId"
              isDisabled={true}
              required={true}
            >
              <label
                className="input-label radio inline-block isDisabled"
                htmlFor="myDomId"
              >
                <Typography
                  color=""
                  variant="field-label"
                >
                  <span
                    className="typography typography-field-label"
                  >
                    Radio Button Card 1
                  </span>
                </Typography>
              </label>
            </Label>
            <HelperText>
              <div
                className="helper-text"
              >
                <Typography
                  variant="helper"
                >
                  <span
                    className="typography typography-helper"
                  >
                    Optional helper text
                  </span>
                </Typography>
              </div>
            </HelperText>
          </div>
        </div>
      </RadioButton>
    </form>
  </Form>
</Formik>
`;
